File: python/nbody/io_utils.py
import csv
from typing import List
from .body import Body

def write_state(writer: csv.writer, step: int, bodies: List[Body]) -> None:
    for i, b in enumerate(bodies):
        writer.writerow([
            step,
            i,
            b.position[0],
            b.position[1],
            b.velocity[0],
            b.velocity[1]
        ])

File: rust/src/io_utils.rs
use std::fs::File;
use std::io::{BufWriter, Write};

use crate::body::Body;

pub fn write_state(
    writer: &mut BufWriter<File>,
    step: usize,
    bodies: &[Body],
) -> std::io::Result<()> {
    for (i, b) in bodies.iter().enumerate() {
        writeln!(
            writer,
            "{},{},{},{},{},{}",
            step,
            i,
            b.position[0],
            b.position[1],
            b.velocity[0],
            b.velocity[1]
        )?;
    }
    Ok(())
}